{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f0f83be6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[32m\u001b[1m    Updating\u001b[22m\u001b[39m git-repo `https://github.com/reworkhow/JWAS.jl.git`\n",
      "\u001b[32m\u001b[1m   Resolving\u001b[22m\u001b[39m package versions...\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Project.toml`\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Manifest.toml`\n",
      "\u001b[32m\u001b[1m    Updating\u001b[22m\u001b[39m registry at `C:\\Users\\aisha\\.julia\\registries\\General.toml`\n",
      "\u001b[32m\u001b[1m    Updating\u001b[22m\u001b[39m git-repo `https://github.com/reworkhow/JWAS.jl.git`\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Project.toml`\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Manifest.toml`\n",
      "\u001b[32m\u001b[1m   Resolving\u001b[22m\u001b[39m package versions...\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Project.toml`\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Manifest.toml`\n",
      "\u001b[32m\u001b[1m   Resolving\u001b[22m\u001b[39m package versions...\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Project.toml`\n",
      "\u001b[32m\u001b[1m  No Changes\u001b[22m\u001b[39m to `C:\\Users\\aisha\\.julia\\environments\\v1.10\\Manifest.toml`\n"
     ]
    }
   ],
   "source": [
    "using Pkg\n",
    "Pkg.add(PackageSpec(name=\"JWAS\", rev=\"master\"))\n",
    "Pkg.update()\n",
    "Pkg.add(\"DataFrames\")\n",
    "Pkg.add(\"CSV\")\n",
    "using JWAS,JWAS.Datasets,DataFrames,CSV,Statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "aec09a08",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[33m\u001b[1m┌ \u001b[22m\u001b[39m\u001b[33m\u001b[1mWarning: \u001b[22m\u001b[39m`missingstrings` keyword argument is deprecated; pass a `Vector{String}` to `missingstring` instead\n",
      "\u001b[33m\u001b[1m└ \u001b[22m\u001b[39m\u001b[90m@ CSV C:\\Users\\aisha\\.julia\\packages\\CSV\\aoJqo\\src\\context.jl:344\u001b[39m\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div><div style = \"float: left;\"><span>5×23 DataFrame</span></div><div style = \"clear: both;\"></div></div><div class = \"data-frame\" style = \"overflow-x: scroll;\"><table class = \"data-frame\" style = \"margin-bottom: 6px;\"><thead><tr class = \"header\"><th class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\">Row</th><th style = \"text-align: left;\">ID</th><th style = \"text-align: left;\">Batch</th><th style = \"text-align: left;\">Company</th><th style = \"text-align: left;\">SowID</th><th style = \"text-align: left;\">EntryAge</th><th style = \"text-align: left;\">Cortisol</th><th style = \"text-align: left;\">Hair1_GrindingDate</th><th style = \"text-align: left;\">C_ExtractionDate</th><th style = \"text-align: left;\">C_GrindingToExtractionTime</th><th style = \"text-align: left;\">qNurlogCort</th><th style = \"text-align: left;\">SW_144836353</th><th style = \"text-align: left;\">SW_144841591</th><th style = \"text-align: left;\">SW_144792892</th><th style = \"text-align: left;\">SW_144876147</th><th style = \"text-align: left;\">SW_144827094</th><th style = \"text-align: left;\">SW_144841166</th><th style = \"text-align: left;\">qNurPenBatch</th><th style = \"text-align: left;\">SW_144783524</th><th style = \"text-align: left;\">SW_144783527</th><th style = \"text-align: left;\">cNurlogCort</th><th style = \"text-align: left;\">cNurPenBatch</th><th style = \"text-align: left;\">GrindingDate2</th><th style = \"text-align: left;\">cNur_groundstorage</th></tr><tr class = \"subheader headerLastRow\"><th class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\"></th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Union{Missing, Int64}\" style = \"text-align: left;\">Int64?</th><th title = \"Union{Missing, String1}\" style = \"text-align: left;\">String1?</th><th title = \"Union{Missing, String15}\" style = \"text-align: left;\">String15?</th><th title = \"Union{Missing, Int64}\" style = \"text-align: left;\">Int64?</th><th title = \"Union{Missing, Float64}\" style = \"text-align: left;\">Float64?</th><th title = \"Union{Missing, String15}\" style = \"text-align: left;\">String15?</th><th title = \"Union{Missing, String15}\" style = \"text-align: left;\">String15?</th><th title = \"Union{Missing, Int64}\" style = \"text-align: left;\">Int64?</th><th title = \"Float64\" style = \"text-align: left;\">Float64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Union{Missing, String7}\" style = \"text-align: left;\">String7?</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Int64\" style = \"text-align: left;\">Int64</th><th title = \"Union{Missing, Float64}\" style = \"text-align: left;\">Float64?</th><th title = \"Union{Missing, String7}\" style = \"text-align: left;\">String7?</th><th title = \"Union{Missing, String15}\" style = \"text-align: left;\">String15?</th><th title = \"Union{Missing, Int64}\" style = \"text-align: left;\">Int64?</th></tr></thead><tbody><tr><td class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\">1</td><td style = \"text-align: right;\">4935</td><td style = \"text-align: right;\">81</td><td style = \"text-align: left;\">E</td><td style = \"text-align: left;\">E8B260</td><td style = \"text-align: right;\">22</td><td style = \"text-align: right;\">16.499</td><td style = \"text-align: left;\">4/20/2022</td><td style = \"text-align: left;\">5/5/2022</td><td style = \"text-align: right;\">15</td><td style = \"text-align: right;\">2.8033</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: left;\">81_1</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">2.14908</td><td style = \"text-align: left;\">81_D02</td><td style = \"text-align: left;\">29/08/2022</td><td style = \"text-align: right;\">85</td></tr><tr><td class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\">2</td><td style = \"text-align: right;\">4936</td><td style = \"text-align: right;\">81</td><td style = \"text-align: left;\">E</td><td style = \"text-align: left;\">W3W953</td><td style = \"text-align: right;\">22</td><td style = \"text-align: right;\">13.6</td><td style = \"text-align: left;\">4/20/2022</td><td style = \"text-align: left;\">5/5/2022</td><td style = \"text-align: right;\">15</td><td style = \"text-align: right;\">2.61007</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: left;\">81_1</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">1.757</td><td style = \"text-align: left;\">81_B04</td><td style = \"text-align: left;\">29/08/2022</td><td style = \"text-align: right;\">85</td></tr><tr><td class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\">3</td><td style = \"text-align: right;\">4937</td><td style = \"text-align: right;\">81</td><td style = \"text-align: left;\">E</td><td style = \"text-align: left;\">E8B205</td><td style = \"text-align: right;\">17</td><td style = \"text-align: right;\">23.1791</td><td style = \"text-align: left;\">4/20/2022</td><td style = \"text-align: left;\">5/5/2022</td><td style = \"text-align: right;\">15</td><td style = \"text-align: right;\">3.14325</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: left;\">81_1</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">2.40153</td><td style = \"text-align: left;\">81_D08</td><td style = \"text-align: left;\">29/08/2022</td><td style = \"text-align: right;\">85</td></tr><tr><td class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\">4</td><td style = \"text-align: right;\">4938</td><td style = \"text-align: right;\">81</td><td style = \"text-align: left;\">E</td><td style = \"text-align: left;\">E8B411</td><td style = \"text-align: right;\">19</td><td style = \"text-align: right;\">16.9478</td><td style = \"text-align: left;\">6/9/2022</td><td style = \"text-align: left;\">5/5/2022</td><td style = \"font-style: italic; text-align: right;\">missing</td><td style = \"text-align: right;\">2.83014</td><td style = \"text-align: right;\">1</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">1</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">1</td><td style = \"text-align: right;\">1</td><td style = \"text-align: left;\">81_1</td><td style = \"text-align: right;\">1</td><td style = \"text-align: right;\">1</td><td style = \"text-align: right;\">-0.71744</td><td style = \"text-align: left;\">81_B04</td><td style = \"text-align: left;\">30/08/2022</td><td style = \"text-align: right;\">84</td></tr><tr><td class = \"rowNumber\" style = \"font-weight: bold; text-align: right;\">5</td><td style = \"text-align: right;\">4939</td><td style = \"text-align: right;\">81</td><td style = \"text-align: left;\">E</td><td style = \"text-align: left;\">3240</td><td style = \"text-align: right;\">16</td><td style = \"text-align: right;\">13.7074</td><td style = \"text-align: left;\">4/22/2022</td><td style = \"text-align: left;\">6/13/2022</td><td style = \"text-align: right;\">52</td><td style = \"text-align: right;\">2.61794</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: left;\">81_1</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">0</td><td style = \"text-align: right;\">1.46326</td><td style = \"text-align: left;\">81_B06</td><td style = \"text-align: left;\">29/08/2022</td><td style = \"text-align: right;\">135</td></tr></tbody></table></div>"
      ],
      "text/latex": [
       "\\begin{tabular}{r|ccccccccc}\n",
       "\t& ID & Batch & Company & SowID & EntryAge & Cortisol & Hair1\\_GrindingDate & C\\_ExtractionDate & \\\\\n",
       "\t\\hline\n",
       "\t& Int64 & Int64? & String1? & String15? & Int64? & Float64? & String15? & String15? & \\\\\n",
       "\t\\hline\n",
       "\t1 & 4935 & 81 & E & E8B260 & 22 & 16.499 & 4/20/2022 & 5/5/2022 & $\\dots$ \\\\\n",
       "\t2 & 4936 & 81 & E & W3W953 & 22 & 13.6 & 4/20/2022 & 5/5/2022 & $\\dots$ \\\\\n",
       "\t3 & 4937 & 81 & E & E8B205 & 17 & 23.1791 & 4/20/2022 & 5/5/2022 & $\\dots$ \\\\\n",
       "\t4 & 4938 & 81 & E & E8B411 & 19 & 16.9478 & 6/9/2022 & 5/5/2022 & $\\dots$ \\\\\n",
       "\t5 & 4939 & 81 & E & 3240 & 16 & 13.7074 & 4/22/2022 & 6/13/2022 & $\\dots$ \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/plain": [
       "\u001b[1m5×23 DataFrame\u001b[0m\n",
       "\u001b[1m Row \u001b[0m│\u001b[1m ID    \u001b[0m\u001b[1m Batch  \u001b[0m\u001b[1m Company  \u001b[0m\u001b[1m SowID     \u001b[0m\u001b[1m EntryAge \u001b[0m\u001b[1m Cortisol \u001b[0m\u001b[1m Hair1_GrindingD\u001b[0m ⋯\n",
       "     │\u001b[90m Int64 \u001b[0m\u001b[90m Int64? \u001b[0m\u001b[90m String1? \u001b[0m\u001b[90m String15? \u001b[0m\u001b[90m Int64?   \u001b[0m\u001b[90m Float64? \u001b[0m\u001b[90m String15?      \u001b[0m ⋯\n",
       "─────┼──────────────────────────────────────────────────────────────────────────\n",
       "   1 │  4935      81  E         E8B260           22   16.499   4/20/2022       ⋯\n",
       "   2 │  4936      81  E         W3W953           22   13.6     4/20/2022\n",
       "   3 │  4937      81  E         E8B205           17   23.1791  4/20/2022\n",
       "   4 │  4938      81  E         E8B411           19   16.9478  6/9/2022\n",
       "   5 │  4939      81  E         3240             16   13.7074  4/22/2022       ⋯\n",
       "\u001b[36m                                                              17 columns omitted\u001b[0m"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phenofile = dataset(\"C:/Users/aisha/Desktop/Grad_School/ROTATIONS/Jack_Lab/Stress_project/Analyses/SWIM_imputation/QTLmarkers/Hormones.csv\")\n",
    "genofile = dataset(\"C:/Users/aisha/Desktop/Grad_School/ROTATIONS/Jack_Lab/Stress_project/Analyses/GWAS/dummy_genotypes.txt\")\n",
    "phenotypes = CSV.read(phenofile,DataFrame,delim = ',',header=true,missingstrings=[\"NA\"])\n",
    "phenotypes = dropmissing(phenotypes, :qNurlogCort)\n",
    "first(phenotypes,5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "192db19e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mThe delimiterd in dummy_genotypes.txt is ','. \u001b[39m\u001b[32mThe header (marker IDs) is provided in dummy_genotypes.txt.\u001b[39m\n",
      "Genotype informatin:\n",
      "#markers: 4999; #individuals: 4095\n"
     ]
    }
   ],
   "source": [
    "MY_G3=[0.263606 0\n",
    "    0 0.555353]\n",
    "myPi=Dict([1.0; 1.0]=>0.000001,[1.0; 0.0]=>0.000999,[0.0; 1.0]=>0.000999,[0.0; 0.0]=>0.998001)\n",
    "genotypes=get_genotypes(genofile,header=true,separator=',',MY_G3,method=\"BayesB\",quality_control=false, Pi = myPi, estimatePi = false,estimate_scale=false);\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2898fd03",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 3: Build Model Equations\n",
    "MY_model_equations = \"qNurlogCort = Batch + EntryAge  + C_GrindingToExtractionTime + SowID + qNurPenBatch + C_ExtractionDate + Hair1_GrindingDate + genotypes\n",
    "             Cortisol = Batch + EntryAge  + C_GrindingToExtractionTime + SowID + qNurPenBatch + C_ExtractionDate + Hair1_GrindingDate + genotypes\";\n",
    "MY_R= [0.567401 0.568697\n",
    "    0.568697 3.75174]\n",
    "MY_model=build_model(MY_model_equations,MY_R);#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "797fa665",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 4: Set Factors or Covariates\n",
    "set_covariate(MY_model, \"EntryAge\");\n",
    "set_covariate(MY_model, \"C_GrindingToExtractionTime\");\n",
    "\n",
    "set_random(MY_model,\"qNurPenBatch\"); \n",
    "set_random(MY_model,\"SowID\"); \n",
    "set_random(MY_model,\"C_ExtractionDate\"); \n",
    "set_random(MY_model,\"Hair1_GrindingDate\"); "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "12a814fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mThe folder results is created to save results.\u001b[39m\n",
      "\u001b[32mChecking genotypes...\u001b[39m\n",
      "\u001b[32mChecking phenotypes...\u001b[39m\n",
      "\u001b[32mIndividual IDs (strings) are provided in the first column of the phenotypic data.\u001b[39m\n",
      "\u001b[32mPredicted values for individuals of interest will be obtained as the summation of Any[] (Note that genomic data is always included for now).\u001b[39m\u001b[31mMissing values are found in column Batch for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column EntryAge for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column C_GrindingToExtractionTime for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column SowID for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column qNurPenBatch for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column Hair1_GrindingDate for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column Batch for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column EntryAge for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column C_GrindingToExtractionTime for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column SowID for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column qNurPenBatch for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[31mMissing values are found in column Hair1_GrindingDate for some observations.Effects of this variable on such observations are considered as zeros. It will be used in the estimation of this effect. Users may impute missing values before the analysis \u001b[39m\n",
      "\u001b[32mPhenotypes for 863 observations are used in the analysis.These individual IDs are saved in the file IDs_for_individuals_with_phenotypes.txt.\u001b[39m\n",
      "\u001b[32mPrior information for random effect variance is not provided and is generated from the data.\u001b[39m\n",
      "\u001b[32mPrior information for random effect variance is not provided and is generated from the data.\u001b[39m\n",
      "\u001b[32mPrior information for random effect variance is not provided and is generated from the data.\u001b[39m\n",
      "\u001b[32mPrior information for random effect variance is not provided and is generated from the data.\u001b[39m\n",
      "\n",
      "The prior for marker effects covariance matrix is calculated from genetic covariance matrix and Π.\n",
      "The mean of the prior for the marker effects covariance matrix is:\n",
      " 0.132913  0.0\n",
      " 0.0       0.280014\n",
      "\n",
      "\n",
      "\n",
      "\u001b[0m\u001b[1mA Linear Mixed Model was build using model equations:\u001b[22m\n",
      "\n",
      "qNurlogCort = Batch + EntryAge  + C_GrindingToExtractionTime + SowID + qNurPenBatch + C_ExtractionDate + Hair1_GrindingDate + genotypes\n",
      "Cortisol = Batch + EntryAge  + C_GrindingToExtractionTime + SowID + qNurPenBatch + C_ExtractionDate + Hair1_GrindingDate + genotypes\n",
      "\n",
      "\u001b[0m\u001b[1mModel Information:\u001b[22m\n",
      "\n",
      "Term            C/F          F/R            nLevels\n",
      "Batch           factor       fixed               15\n",
      "EntryAge        covariate    fixed                1\n",
      "C_GrindingToExtractionTime covariate    fixed                1\n",
      "SowID           factor       random             281\n",
      "qNurPenBatch    factor       random              68\n",
      "C_ExtractionDate factor       random              22\n",
      "Hair1_GrindingDate factor       random              55\n",
      "\n",
      "\u001b[0m\u001b[1mMCMC Information:\u001b[22m\n",
      "\n",
      "chain_length                                    100\n",
      "burnin                                            0\n",
      "starting_value                                 true\n",
      "printout_frequency                              101\n",
      "output_samples_frequency                         10\n",
      "constraint on residual variance               false\n",
      "constraint on marker effect variance for genotypes false\n",
      "missing_phenotypes                             true\n",
      "update_priors_frequency                           0\n",
      "seed                                          false\n",
      "\n",
      "\u001b[0m\u001b[1mHyper-parameters Information:\u001b[22m\n",
      "\n",
      "random effect variances (qNurlogCort:qNurPenBatch,Cortisol:qNurPenBatch):\n",
      " 0.018  0.0\n",
      " 0.0    5.208\n",
      "random effect variances (qNurlogCort:SowID,Cortisol:SowID):\n",
      " 0.018  0.0\n",
      " 0.0    5.208\n",
      "random effect variances (qNurlogCort:C_ExtractionDate,Cortisol:C_ExtractionDate):\n",
      " 0.018  0.0\n",
      " 0.0    5.208\n",
      "random effect variances (qNurlogCort:Hair1_GrindingDate,Cortisol:Hair1_GrindingDate):\n",
      " 0.018  0.0\n",
      " 0.0    5.208\n",
      "residual variances:           \n",
      " 0.567f0  0.569f0\n",
      " 0.569f0  3.752f0\n",
      "\n",
      "\u001b[0m\u001b[1mGenomic Information:\u001b[22m\n",
      "\n",
      "complete genomic data (i.e., non-single-step analysis)\n",
      "\n",
      "Genomic Category                          genotypes\n",
      "Method                                       BayesB\n",
      "genetic variances (genomic):  \n",
      " 0.264  0.0\n",
      " 0.0    0.555\n",
      "marker effect variances:      \n",
      " 0.133  0.0\n",
      " 0.0    0.28\n",
      "\n",
      "Π: (Y(yes):included; N(no):excluded)\n",
      "\n",
      "[\"qNurlogCort\", \"Cortisol\"]         probability\n",
      "[\"N\", \"N\"]            0.998001\n",
      "[\"Y\", \"Y\"]              1.0e-6\n",
      "[\"Y\", \"N\"]            0.000999\n",
      "[\"N\", \"Y\"]            0.000999\n",
      "\n",
      "estimatePi                                    false\n",
      "estimate_scale                                false\n",
      "\n",
      "\u001b[0m\u001b[1mDegree of freedom for hyper-parameters:\u001b[22m\n",
      "\n",
      "residual variances:                           6.000\n",
      "random effect variances:                      6.000\n",
      "random effect variances:                      6.000\n",
      "random effect variances:                      6.000\n",
      "random effect variances:                      6.000\n",
      "marker effect variances:                      6.000\n",
      "\n",
      "\n",
      "\n",
      "\u001b[32mThe file results/MCMC_samples_residual_variance.txt is created to save MCMC samples for residual_variance.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_marker_effects_genotypes_qNurlogCort.txt is created to save MCMC samples for marker_effects_genotypes_qNurlogCort.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_marker_effects_genotypes_Cortisol.txt is created to save MCMC samples for marker_effects_genotypes_Cortisol.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_marker_effects_variances_genotypes.txt is created to save MCMC samples for marker_effects_variances_genotypes.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_pi_genotypes.txt is created to save MCMC samples for pi_genotypes.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_qNurlogCort.qNurPenBatch_Cortisol.qNurPenBatch_variances.txt is created to save MCMC samples for qNurlogCort:qNurPenBatch_Cortisol:qNurPenBatch_variances.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_qNurlogCort.SowID_Cortisol.SowID_variances.txt is created to save MCMC samples for qNurlogCort:SowID_Cortisol:SowID_variances.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_qNurlogCort.C_ExtractionDate_Cortisol.C_ExtractionDate_variances.txt is created to save MCMC samples for qNurlogCort:C_ExtractionDate_Cortisol:C_ExtractionDate_variances.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_qNurlogCort.Hair1_GrindingDate_Cortisol.Hair1_GrindingDate_variances.txt is created to save MCMC samples for qNurlogCort:Hair1_GrindingDate_Cortisol:Hair1_GrindingDate_variances.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_EBV_qNurlogCort.txt is created to save MCMC samples for EBV_qNurlogCort.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_EBV_Cortisol.txt is created to save MCMC samples for EBV_Cortisol.\u001b[39m\n",
      "\u001b[32mThe file results/MCMC_samples_genetic_variance.txt is created to save MCMC samples for genetic_variance.\u001b[39m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "\u001b[32mrunning MCMC ...   2%|█                                  |  ETA: 0:05:18\u001b[39m"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mThe file results/MCMC_samples_heritability.txt is created to save MCMC samples for heritability.\u001b[39m\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[32mrunning MCMC ... 100%|███████████████████████████████████| Time: 0:00:14\u001b[39m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\u001b[0m\u001b[1mThe version of Julia and Platform in use:\u001b[22m\n",
      "\n",
      "Julia Version 1.10.0\n",
      "Commit 3120989f39 (2023-12-25 18:01 UTC)\n",
      "Build Info:\n",
      "  Official https://julialang.org/ release\n",
      "Platform Info:\n",
      "  OS: Windows (x86_64-w64-mingw32)\n",
      "  CPU: 16 × AMD Ryzen 7 5700U with Radeon Graphics         \n",
      "  WORD_SIZE: 64\n",
      "  LIBM: libopenlibm\n",
      "  LLVM: libLLVM-15.0.7 (ORCJIT, znver2)\n",
      "  Threads: 2 on 16 virtual cores\n",
      "\n",
      "\n",
      "\u001b[0m\u001b[1mThe analysis has finished. Results are saved in the returned \u001b[22m\u001b[0m\u001b[1mvariable and text files. MCMC samples are saved in text files.\u001b[22m\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "out=runMCMC(MY_model,phenotypes,missing_phenotypes=true,estimate_variance=true,chain_length=100,burnin=0,output_heritability=true,output_samples_frequency=10);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "911cbe97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Compute the posterior probability of association of the genomic window that explains more than 0.001 of the total genetic variance."
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r",
      "\u001b[32mrunning GWAS...  20%|████████                            |  ETA: 0:00:02\u001b[39m"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[32mrunning GWAS... 100%|████████████████████████████████████| Time: 0:00:01\u001b[39m\n",
      "\u001b[32mrunning GWAS... 100%|████████████████████████████████████| Time: 0:00:00\u001b[39m\n",
      "\u001b[32mcalculating genomic correlation... 100%|█████████████████| Time: 0:00:00\u001b[39m\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"C:\\\\Users\\\\aisha\\\\results\""
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cd(\"C:/Users/aisha/results\")\n",
    "out=GWAS(MY_model,\"dummy_mapfile.txt\",\"MCMC_samples_marker_effects_genotypes_Cortisol.txt\",\"MCMC_samples_marker_effects_genotypes_qNurlogCort.txt\",header=true,window_size=\"0.25 Mb\",sliding_window=false,GWAS=true,genetic_correlation=true)\n",
    "#sort(out, :estimate_cor, rev=true)\n",
    "#CSV.write(\"winVarPropsBivar.txt\",out)\n",
    "pwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5f87b48b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "search: \u001b[0m\u001b[1mr\u001b[22m\u001b[0m\u001b[1mu\u001b[22m\u001b[0m\u001b[1mn\u001b[22m\u001b[0m\u001b[1mM\u001b[22m\u001b[0m\u001b[1mC\u001b[22m\u001b[0m\u001b[1mM\u001b[22m\u001b[0m\u001b[1mC\u001b[22m\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/latex": [
       "\\begin{verbatim}\n",
       "runMCMC(model::MME,df::DataFrame;\n",
       "        #Data\n",
       "        heterogeneous_residuals           = false,\n",
       "        #MCMC\n",
       "        chain_length::Integer             = 100,\n",
       "        starting_value                    = false,\n",
       "        burnin::Integer                   = 0,\n",
       "        output_samples_frequency::Integer = chain_length>1000 ? div(chain_length,1000) : 1,\n",
       "        update_priors_frequency::Integer  = 0,\n",
       "        #Methods\n",
       "        single_step_analysis            = false, #parameters for single-step analysis\n",
       "        pedigree                        = false, #parameters for single-step analysis\n",
       "        fitting_J_vector                = true,  #parameters for single-step analysis\n",
       "        categorical_trait               = false,\n",
       "        censored_trait                  = false,\n",
       "        causal_structure                = false,\n",
       "        mega_trait                      = false,\n",
       "        missing_phenotypes              = true,\n",
       "        constraint                      = false,\n",
       "        #Genomic Prediction\n",
       "        outputEBV                       = true,\n",
       "        output_heritability             = true,\n",
       "        prediction_equation             = false,\n",
       "        #MISC\n",
       "        seed                            = false,\n",
       "        printout_model_info             = true,\n",
       "        printout_frequency              = chain_length+1,\n",
       "        big_memory                      = false,\n",
       "        double_precision                = false,\n",
       "        ##MCMC samples (defaut to marker effects and hyperparametes (variance components))\n",
       "        output_folder                     = \"results\",\n",
       "        output_samples_for_all_parameters = false,\n",
       "        ##for deprecated JWAS\n",
       "        methods                         = \"conventional (no markers)\",\n",
       "        Pi                              = 0.0,\n",
       "        estimatePi                      = false)\n",
       "\\end{verbatim}\n",
       "\\textbf{Run MCMC for Bayesian Linear Mixed Models with or without estimation of variance components.}\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Markov chain Monte Carlo\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item The first \\texttt{burnin} iterations are discarded at the beginning of a MCMC chain of length \\texttt{chain\\_length}.\n",
       "\n",
       "\n",
       "\\item Save MCMC samples every \\texttt{output\\_samples\\_frequency} iterations, defaulting to \\texttt{chain\\_length/1000}, to a folder \\texttt{output\\_folder}, defaulting to \\texttt{results}. MCMC samples for hyperparametes (variance componets) and marker effects are saved by default. MCMC samples for location parametes can be saved using function \\texttt{output\\_MCMC\\_samples()}. Note that saving MCMC samples too frequently slows down the computation.\n",
       "\n",
       "\n",
       "\\item The \\texttt{starting\\_value} can be provided as a vector for all location parameteres and marker effects, defaulting to \\texttt{0.0}s. The order of starting values for location parameters and marker effects should be the order of location parameters in the Mixed Model Equation for all traits (This can be obtained by getNames(model)) and then markers for all traits (all markers for trait 1 then all markers for trait 2...).\n",
       "\n",
       "\n",
       "\\item Miscellaneous Options\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Priors are updated every \\texttt{update\\_priors\\_frequency} iterations, defaulting to \\texttt{0}.\n",
       "\n",
       "\\end{itemize}\n",
       "\\end{itemize}\n",
       "\n",
       "\\item Methods\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Single step analysis is allowed if \\texttt{single\\_step\\_analysis} = \\texttt{true} and \\texttt{pedigree} is provided.\n",
       "\n",
       "\n",
       "\\item Miscellaneous Options\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Missing phenotypes are allowed in multi-trait analysis with \\texttt{missing\\_phenotypes}=true, defaulting to \\texttt{true}.\n",
       "\n",
       "\n",
       "\\item Catogorical Traits are allowed if \\texttt{categorical\\_trait}=true, defaulting to \\texttt{false}. Phenotypes should be coded as 1,2,3...\n",
       "\n",
       "\n",
       "\\item Censored traits are allowed if the upper bounds are provided in \\texttt{censored\\_trait} as an array, and lower bounds are provided as phenotypes.\n",
       "\n",
       "\n",
       "\\item If \\texttt{constraint}=true, defaulting to \\texttt{false}, constrain residual covariances between traits to be zeros.\n",
       "\n",
       "\n",
       "\\item If \\texttt{causal\\_structure} is provided, e.g., causal\\_structure = [0.0 0.0 0.0;1.0 0.0 0.0;1.0 0.0 0.0] for trait 1 -> trait 2 and trait 1 -> trait 3 (column index affacts row index, and a lower triangular matrix is required), phenotypic causal networks will be incorporated using structure equation models.\n",
       "\n",
       "\\end{itemize}\n",
       "\\end{itemize}\n",
       "\n",
       "\\item Genomic Prediction\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Predicted values for individuals of interest can be obtained based on a user-defined prediction equation \\texttt{prediction\\_equation}, e.g., \"y1:animal + y1:age\".\n",
       "\n",
       "\\end{itemize}\n",
       "For now, genomic data is always included. Genetic values including effects defined with genotype and pedigree information are returned if \\texttt{prediction\\_equation}= false, defaulting to \\texttt{false}.\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Individual estimted genetic values and prediction error variances (PEVs) are returned if \\texttt{outputEBV}=true, defaulting to \\texttt{true}. Heritability and genetic\n",
       "\n",
       "\\end{itemize}\n",
       "variances are returned if \\texttt{output\\_heritability}=\\texttt{true}, defaulting to \\texttt{true}. Note that estimation of heritability is computaionally intensive.\n",
       "\n",
       "\n",
       "\\item Miscellaneous Options\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item Print out the model information in REPL if \\texttt{printout\\_model\\_info}=true; print out the monte carlo mean in REPL with \\texttt{printout\\_frequency}, defaulting to \\texttt{false}.\n",
       "\n",
       "\n",
       "\\item If \\texttt{seed}, defaulting to \\texttt{false}, is provided, a reproducible sequence of numbers will be generated for random number generation.\n",
       "\n",
       "\n",
       "\\item If \\texttt{big\\_memory}=true, defaulting to \\texttt{false}, a machine with  lots of memory is assumed which may speed up the analysis.\n",
       "\n",
       "\\end{itemize}\n",
       "\\end{itemize}\n"
      ],
      "text/markdown": [
       "```\n",
       "runMCMC(model::MME,df::DataFrame;\n",
       "        #Data\n",
       "        heterogeneous_residuals           = false,\n",
       "        #MCMC\n",
       "        chain_length::Integer             = 100,\n",
       "        starting_value                    = false,\n",
       "        burnin::Integer                   = 0,\n",
       "        output_samples_frequency::Integer = chain_length>1000 ? div(chain_length,1000) : 1,\n",
       "        update_priors_frequency::Integer  = 0,\n",
       "        #Methods\n",
       "        single_step_analysis            = false, #parameters for single-step analysis\n",
       "        pedigree                        = false, #parameters for single-step analysis\n",
       "        fitting_J_vector                = true,  #parameters for single-step analysis\n",
       "        categorical_trait               = false,\n",
       "        censored_trait                  = false,\n",
       "        causal_structure                = false,\n",
       "        mega_trait                      = false,\n",
       "        missing_phenotypes              = true,\n",
       "        constraint                      = false,\n",
       "        #Genomic Prediction\n",
       "        outputEBV                       = true,\n",
       "        output_heritability             = true,\n",
       "        prediction_equation             = false,\n",
       "        #MISC\n",
       "        seed                            = false,\n",
       "        printout_model_info             = true,\n",
       "        printout_frequency              = chain_length+1,\n",
       "        big_memory                      = false,\n",
       "        double_precision                = false,\n",
       "        ##MCMC samples (defaut to marker effects and hyperparametes (variance components))\n",
       "        output_folder                     = \"results\",\n",
       "        output_samples_for_all_parameters = false,\n",
       "        ##for deprecated JWAS\n",
       "        methods                         = \"conventional (no markers)\",\n",
       "        Pi                              = 0.0,\n",
       "        estimatePi                      = false)\n",
       "```\n",
       "\n",
       "**Run MCMC for Bayesian Linear Mixed Models with or without estimation of variance components.**\n",
       "\n",
       "  * Markov chain Monte Carlo\n",
       "\n",
       "      * The first `burnin` iterations are discarded at the beginning of a MCMC chain of length `chain_length`.\n",
       "      * Save MCMC samples every `output_samples_frequency` iterations, defaulting to `chain_length/1000`, to a folder `output_folder`, defaulting to `results`. MCMC samples for hyperparametes (variance componets) and marker effects are saved by default. MCMC samples for location parametes can be saved using function `output_MCMC_samples()`. Note that saving MCMC samples too frequently slows down the computation.\n",
       "      * The `starting_value` can be provided as a vector for all location parameteres and marker effects, defaulting to `0.0`s. The order of starting values for location parameters and marker effects should be the order of location parameters in the Mixed Model Equation for all traits (This can be obtained by getNames(model)) and then markers for all traits (all markers for trait 1 then all markers for trait 2...).\n",
       "      * Miscellaneous Options\n",
       "\n",
       "          * Priors are updated every `update_priors_frequency` iterations, defaulting to `0`.\n",
       "  * Methods\n",
       "\n",
       "      * Single step analysis is allowed if `single_step_analysis` = `true` and `pedigree` is provided.\n",
       "      * Miscellaneous Options\n",
       "\n",
       "          * Missing phenotypes are allowed in multi-trait analysis with `missing_phenotypes`=true, defaulting to `true`.\n",
       "          * Catogorical Traits are allowed if `categorical_trait`=true, defaulting to `false`. Phenotypes should be coded as 1,2,3...\n",
       "          * Censored traits are allowed if the upper bounds are provided in `censored_trait` as an array, and lower bounds are provided as phenotypes.\n",
       "          * If `constraint`=true, defaulting to `false`, constrain residual covariances between traits to be zeros.\n",
       "          * If `causal_structure` is provided, e.g., causal_structure = [0.0 0.0 0.0;1.0 0.0 0.0;1.0 0.0 0.0] for trait 1 -> trait 2 and trait 1 -> trait 3 (column index affacts row index, and a lower triangular matrix is required), phenotypic causal networks will be incorporated using structure equation models.\n",
       "  * Genomic Prediction\n",
       "\n",
       "      * Predicted values for individuals of interest can be obtained based on a user-defined prediction equation `prediction_equation`, e.g., \"y1:animal + y1:age\".\n",
       "\n",
       "    For now, genomic data is always included. Genetic values including effects defined with genotype and pedigree information are returned if `prediction_equation`= false, defaulting to `false`.\n",
       "\n",
       "      * Individual estimted genetic values and prediction error variances (PEVs) are returned if `outputEBV`=true, defaulting to `true`. Heritability and genetic\n",
       "\n",
       "    variances are returned if `output_heritability`=`true`, defaulting to `true`. Note that estimation of heritability is computaionally intensive.\n",
       "  * Miscellaneous Options\n",
       "\n",
       "      * Print out the model information in REPL if `printout_model_info`=true; print out the monte carlo mean in REPL with `printout_frequency`, defaulting to `false`.\n",
       "      * If `seed`, defaulting to `false`, is provided, a reproducible sequence of numbers will be generated for random number generation.\n",
       "      * If `big_memory`=true, defaulting to `false`, a machine with  lots of memory is assumed which may speed up the analysis.\n"
      ],
      "text/plain": [
       "\u001b[36m  runMCMC(model::MME,df::DataFrame;\u001b[39m\n",
       "\u001b[36m          #Data\u001b[39m\n",
       "\u001b[36m          heterogeneous_residuals           = false,\u001b[39m\n",
       "\u001b[36m          #MCMC\u001b[39m\n",
       "\u001b[36m          chain_length::Integer             = 100,\u001b[39m\n",
       "\u001b[36m          starting_value                    = false,\u001b[39m\n",
       "\u001b[36m          burnin::Integer                   = 0,\u001b[39m\n",
       "\u001b[36m          output_samples_frequency::Integer = chain_length>1000 ? div(chain_length,1000) : 1,\u001b[39m\n",
       "\u001b[36m          update_priors_frequency::Integer  = 0,\u001b[39m\n",
       "\u001b[36m          #Methods\u001b[39m\n",
       "\u001b[36m          single_step_analysis            = false, #parameters for single-step analysis\u001b[39m\n",
       "\u001b[36m          pedigree                        = false, #parameters for single-step analysis\u001b[39m\n",
       "\u001b[36m          fitting_J_vector                = true,  #parameters for single-step analysis\u001b[39m\n",
       "\u001b[36m          categorical_trait               = false,\u001b[39m\n",
       "\u001b[36m          censored_trait                  = false,\u001b[39m\n",
       "\u001b[36m          causal_structure                = false,\u001b[39m\n",
       "\u001b[36m          mega_trait                      = false,\u001b[39m\n",
       "\u001b[36m          missing_phenotypes              = true,\u001b[39m\n",
       "\u001b[36m          constraint                      = false,\u001b[39m\n",
       "\u001b[36m          #Genomic Prediction\u001b[39m\n",
       "\u001b[36m          outputEBV                       = true,\u001b[39m\n",
       "\u001b[36m          output_heritability             = true,\u001b[39m\n",
       "\u001b[36m          prediction_equation             = false,\u001b[39m\n",
       "\u001b[36m          #MISC\u001b[39m\n",
       "\u001b[36m          seed                            = false,\u001b[39m\n",
       "\u001b[36m          printout_model_info             = true,\u001b[39m\n",
       "\u001b[36m          printout_frequency              = chain_length+1,\u001b[39m\n",
       "\u001b[36m          big_memory                      = false,\u001b[39m\n",
       "\u001b[36m          double_precision                = false,\u001b[39m\n",
       "\u001b[36m          ##MCMC samples (defaut to marker effects and hyperparametes (variance components))\u001b[39m\n",
       "\u001b[36m          output_folder                     = \"results\",\u001b[39m\n",
       "\u001b[36m          output_samples_for_all_parameters = false,\u001b[39m\n",
       "\u001b[36m          ##for deprecated JWAS\u001b[39m\n",
       "\u001b[36m          methods                         = \"conventional (no markers)\",\u001b[39m\n",
       "\u001b[36m          Pi                              = 0.0,\u001b[39m\n",
       "\u001b[36m          estimatePi                      = false)\u001b[39m\n",
       "\n",
       "  \u001b[1mRun MCMC for Bayesian Linear Mixed Models with or without estimation of\n",
       "  variance components.\u001b[22m\n",
       "\n",
       "    •  Markov chain Monte Carlo\n",
       "       • The first \u001b[36mburnin\u001b[39m iterations are discarded at the\n",
       "       beginning of a MCMC chain of length \u001b[36mchain_length\u001b[39m.\n",
       "       • Save MCMC samples every \u001b[36moutput_samples_frequency\u001b[39m\n",
       "       iterations, defaulting to \u001b[36mchain_length/1000\u001b[39m, to a folder\n",
       "       \u001b[36moutput_folder\u001b[39m, defaulting to \u001b[36mresults\u001b[39m. MCMC samples for\n",
       "       hyperparametes (variance componets) and marker effects\n",
       "       are saved by default. MCMC samples for location\n",
       "       parametes can be saved using function\n",
       "       \u001b[36moutput_MCMC_samples()\u001b[39m. Note that saving MCMC samples too\n",
       "       frequently slows down the computation.\n",
       "       • The \u001b[36mstarting_value\u001b[39m can be provided as a vector for all\n",
       "       location parameteres and marker effects, defaulting to\n",
       "       \u001b[36m0.0\u001b[39ms. The order of starting values for location\n",
       "       parameters and marker effects should be the order of\n",
       "       location parameters in the Mixed Model Equation for all\n",
       "       traits (This can be obtained by getNames(model)) and\n",
       "       then markers for all traits (all markers for trait 1\n",
       "       then all markers for trait 2...).\n",
       "       • Miscellaneous Options\n",
       "       • Priors are updated every\n",
       "       \u001b[36mupdate_priors_frequency\u001b[39m iterations, defaulting\n",
       "       to \u001b[36m0\u001b[39m.\n",
       "\n",
       "    •  Methods\n",
       "       • Single step analysis is allowed if \u001b[36msingle_step_analysis\u001b[39m\n",
       "       = \u001b[36mtrue\u001b[39m and \u001b[36mpedigree\u001b[39m is provided.\n",
       "       • Miscellaneous Options\n",
       "       • Missing phenotypes are allowed in multi-trait\n",
       "       analysis with \u001b[36mmissing_phenotypes\u001b[39m=true,\n",
       "       defaulting to \u001b[36mtrue\u001b[39m.\n",
       "       • Catogorical Traits are allowed if\n",
       "       \u001b[36mcategorical_trait\u001b[39m=true, defaulting to \u001b[36mfalse\u001b[39m.\n",
       "       Phenotypes should be coded as 1,2,3...\n",
       "       • Censored traits are allowed if the upper\n",
       "       bounds are provided in \u001b[36mcensored_trait\u001b[39m as an\n",
       "       array, and lower bounds are provided as\n",
       "       phenotypes.\n",
       "       • If \u001b[36mconstraint\u001b[39m=true, defaulting to \u001b[36mfalse\u001b[39m,\n",
       "       constrain residual covariances between traits\n",
       "       to be zeros.\n",
       "       • If \u001b[36mcausal_structure\u001b[39m is provided, e.g.,\n",
       "       causal_structure = [0.0 0.0 0.0;1.0 0.0\n",
       "       0.0;1.0 0.0 0.0] for trait 1 -> trait 2 and\n",
       "       trait 1 -> trait 3 (column index affacts row\n",
       "       index, and a lower triangular matrix is\n",
       "       required), phenotypic causal networks will be\n",
       "       incorporated using structure equation models.\n",
       "\n",
       "    •  Genomic Prediction\n",
       "       • Predicted values for individuals of interest can be\n",
       "       obtained based on a user-defined prediction equation\n",
       "       \u001b[36mprediction_equation\u001b[39m, e.g., \"y1:animal + y1:age\".\n",
       "       For now, genomic data is always included. Genetic values including\n",
       "       effects defined with genotype and pedigree information are\n",
       "       returned if \u001b[36mprediction_equation\u001b[39m= false, defaulting to \u001b[36mfalse\u001b[39m.\n",
       "       • Individual estimted genetic values and prediction error\n",
       "       variances (PEVs) are returned if \u001b[36moutputEBV\u001b[39m=true,\n",
       "       defaulting to \u001b[36mtrue\u001b[39m. Heritability and genetic\n",
       "       variances are returned if \u001b[36moutput_heritability\u001b[39m=\u001b[36mtrue\u001b[39m, defaulting to\n",
       "       \u001b[36mtrue\u001b[39m. Note that estimation of heritability is computaionally\n",
       "       intensive.\n",
       "\n",
       "    •  Miscellaneous Options\n",
       "       • Print out the model information in REPL if\n",
       "       \u001b[36mprintout_model_info\u001b[39m=true; print out the monte carlo mean\n",
       "       in REPL with \u001b[36mprintout_frequency\u001b[39m, defaulting to \u001b[36mfalse\u001b[39m.\n",
       "       • If \u001b[36mseed\u001b[39m, defaulting to \u001b[36mfalse\u001b[39m, is provided, a\n",
       "       reproducible sequence of numbers will be generated for\n",
       "       random number generation.\n",
       "       • If \u001b[36mbig_memory\u001b[39m=true, defaulting to \u001b[36mfalse\u001b[39m, a machine with\n",
       "       lots of memory is assumed which may speed up the\n",
       "       analysis."
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "?runMCMC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "005e7d72",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "search: \u001b[0m\u001b[1mG\u001b[22m\u001b[0m\u001b[1mW\u001b[22m\u001b[0m\u001b[1mA\u001b[22m\u001b[0m\u001b[1mS\u001b[22m\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/latex": [
       "\\begin{verbatim}\n",
       "GWAS(marker_effects_file,model;header=true,window_size=100,threshold=0.001)\n",
       "\\end{verbatim}\n",
       "run genomic window-based GWAS without a map file\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item MCMC samples of marker effects are stored in \\textbf{marker\\emph{effects}file} with delimiter ','.\n",
       "\n",
       "\n",
       "\\item \\textbf{window\\_size} is either a constant (identical number of markers in each window) or an array of number of markers in each window\n",
       "\n",
       "\n",
       "\\item \\textbf{model} is either the model::MME used in analysis or the genotypic covariate matrix M::Array\n",
       "\n",
       "\n",
       "\\item File format:\n",
       "\n",
       "\\end{itemize}\n",
       "\\rule{\\textwidth}{1pt}\n",
       "\\begin{verbatim}\n",
       "GWAS(marker_effects_file;header=true)\n",
       "\\end{verbatim}\n",
       "Compute the model frequency for each marker (the probability the marker is included in the model) using samples of marker effects stored in \\textbf{marker\\emph{effects}file}.\n",
       "\n",
       "\\rule{\\textwidth}{1pt}\n",
       "\\begin{verbatim}\n",
       "GWAS(model,map_file,marker_effects_file...;\n",
       "     window_size = \"1 Mb\",sliding_window = false,\n",
       "     GWAS = true, threshold = 0.001,\n",
       "     genetic_correlation = false,\n",
       "     header = true)\n",
       "\\end{verbatim}\n",
       "run genomic window-based GWAS\n",
       "\n",
       "\\begin{itemize}\n",
       "\\item MCMC samples of marker effects are stored in \\textbf{marker\\emph{effects}file} with delimiter ','.\n",
       "\n",
       "\n",
       "\\item \\textbf{model} is either the model::MME used in analysis or the genotype cavariate matrix M::Array\n",
       "\n",
       "\n",
       "\\item \\textbf{map\\_file} has the (sorted) marker position information with delimiter ','. If the map file is not provided, i.e., \\textbf{map\\_file}=\\texttt{false}, a fake map file will be generated with \\textbf{window\\_size} markers in each 1 Mb window, and each 1 Mb window will be tested.\n",
       "\n",
       "\n",
       "\\item If two \\textbf{marker\\emph{effects}file} are provided, and \\textbf{genetic\\_correlation} = true, genomic correlation for each window is calculated.\n",
       "\n",
       "\n",
       "\\item Statistics are computed for nonoverlapping windows of size \\emph{window\\_size} by default. If \\textbf{sliding\\_window} = true, those for overlapping sliding windows are calculated.\n",
       "\n",
       "\n",
       "\\item map file format:\n",
       "\n",
       "\\end{itemize}\n",
       "\\begin{verbatim}\n",
       "markerID,chromosome,position\n",
       "m1,1,16977\n",
       "m2,1,434311\n",
       "m3,1,1025513\n",
       "m4,2,70350\n",
       "m5,2,101135\n",
       "\\end{verbatim}\n"
      ],
      "text/markdown": [
       "```\n",
       "GWAS(marker_effects_file,model;header=true,window_size=100,threshold=0.001)\n",
       "```\n",
       "\n",
       "run genomic window-based GWAS without a map file\n",
       "\n",
       "  * MCMC samples of marker effects are stored in **marker*effects*file** with delimiter ','.\n",
       "  * **window_size** is either a constant (identical number of markers in each window) or an array of number of markers in each window\n",
       "  * **model** is either the model::MME used in analysis or the genotypic covariate matrix M::Array\n",
       "  * File format:\n",
       "\n",
       "---\n",
       "\n",
       "```\n",
       "GWAS(marker_effects_file;header=true)\n",
       "```\n",
       "\n",
       "Compute the model frequency for each marker (the probability the marker is included in the model) using samples of marker effects stored in **marker*effects*file**.\n",
       "\n",
       "---\n",
       "\n",
       "```\n",
       "GWAS(model,map_file,marker_effects_file...;\n",
       "     window_size = \"1 Mb\",sliding_window = false,\n",
       "     GWAS = true, threshold = 0.001,\n",
       "     genetic_correlation = false,\n",
       "     header = true)\n",
       "```\n",
       "\n",
       "run genomic window-based GWAS\n",
       "\n",
       "  * MCMC samples of marker effects are stored in **marker*effects*file** with delimiter ','.\n",
       "  * **model** is either the model::MME used in analysis or the genotype cavariate matrix M::Array\n",
       "  * **map_file** has the (sorted) marker position information with delimiter ','. If the map file is not provided, i.e., **map_file**=`false`, a fake map file will be generated with **window_size** markers in each 1 Mb window, and each 1 Mb window will be tested.\n",
       "  * If two **marker*effects*file** are provided, and **genetic_correlation** = true, genomic correlation for each window is calculated.\n",
       "  * Statistics are computed for nonoverlapping windows of size *window_size* by default. If **sliding_window** = true, those for overlapping sliding windows are calculated.\n",
       "  * map file format:\n",
       "\n",
       "```\n",
       "markerID,chromosome,position\n",
       "m1,1,16977\n",
       "m2,1,434311\n",
       "m3,1,1025513\n",
       "m4,2,70350\n",
       "m5,2,101135\n",
       "```\n"
      ],
      "text/plain": [
       "\u001b[36m  GWAS(marker_effects_file,model;header=true,window_size=100,threshold=0.001)\u001b[39m\n",
       "\n",
       "  run genomic window-based GWAS without a map file\n",
       "\n",
       "    •  MCMC samples of marker effects are stored in \u001b[1mmarker\u001b[4meffects\u001b[24mfile\u001b[22m\n",
       "       with delimiter ','.\n",
       "\n",
       "    •  \u001b[1mwindow_size\u001b[22m is either a constant (identical number of markers in\n",
       "       each window) or an array of number of markers in each window\n",
       "\n",
       "    •  \u001b[1mmodel\u001b[22m is either the model::MME used in analysis or the genotypic\n",
       "       covariate matrix M::Array\n",
       "\n",
       "    •  File format:\n",
       "\n",
       "  ────────────────────────────────────────────────────────────────────────────\n",
       "\n",
       "\u001b[36m  GWAS(marker_effects_file;header=true)\u001b[39m\n",
       "\n",
       "  Compute the model frequency for each marker (the probability the marker is\n",
       "  included in the model) using samples of marker effects stored in\n",
       "  \u001b[1mmarker\u001b[4meffects\u001b[24mfile\u001b[22m.\n",
       "\n",
       "  ────────────────────────────────────────────────────────────────────────────\n",
       "\n",
       "\u001b[36m  GWAS(model,map_file,marker_effects_file...;\u001b[39m\n",
       "\u001b[36m       window_size = \"1 Mb\",sliding_window = false,\u001b[39m\n",
       "\u001b[36m       GWAS = true, threshold = 0.001,\u001b[39m\n",
       "\u001b[36m       genetic_correlation = false,\u001b[39m\n",
       "\u001b[36m       header = true)\u001b[39m\n",
       "\n",
       "  run genomic window-based GWAS\n",
       "\n",
       "    •  MCMC samples of marker effects are stored in \u001b[1mmarker\u001b[4meffects\u001b[24mfile\u001b[22m\n",
       "       with delimiter ','.\n",
       "\n",
       "    •  \u001b[1mmodel\u001b[22m is either the model::MME used in analysis or the genotype\n",
       "       cavariate matrix M::Array\n",
       "\n",
       "    •  \u001b[1mmap_file\u001b[22m has the (sorted) marker position information with\n",
       "       delimiter ','. If the map file is not provided, i.e.,\n",
       "       \u001b[1mmap_file\u001b[22m=\u001b[36mfalse\u001b[39m, a fake map file will be generated with \u001b[1mwindow_size\u001b[22m\n",
       "       markers in each 1 Mb window, and each 1 Mb window will be tested.\n",
       "\n",
       "    •  If two \u001b[1mmarker\u001b[4meffects\u001b[24mfile\u001b[22m are provided, and \u001b[1mgenetic_correlation\u001b[22m =\n",
       "       true, genomic correlation for each window is calculated.\n",
       "\n",
       "    •  Statistics are computed for nonoverlapping windows of size\n",
       "       \u001b[4mwindow_size\u001b[24m by default. If \u001b[1msliding_window\u001b[22m = true, those for\n",
       "       overlapping sliding windows are calculated.\n",
       "\n",
       "    •  map file format:\n",
       "\n",
       "\u001b[36m  markerID,chromosome,position\u001b[39m\n",
       "\u001b[36m  m1,1,16977\u001b[39m\n",
       "\u001b[36m  m2,1,434311\u001b[39m\n",
       "\u001b[36m  m3,1,1025513\u001b[39m\n",
       "\u001b[36m  m4,2,70350\u001b[39m\n",
       "\u001b[36m  m5,2,101135\u001b[39m"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "?GWAS"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Julia 1.10.0",
   "language": "julia",
   "name": "julia-1.10"
  },
  "language_info": {
   "file_extension": ".jl",
   "mimetype": "application/julia",
   "name": "julia",
   "version": "1.10.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
